backend/portal/logic.py

```diff
@@ -6,26 +9,60 @@
 User = get_user_model()
 
 
+def get_user_info(user):
+    """Returns Platform user information"""
+    response = authenticated_request(user, "GET", "https://platform.pennlabs.org/accounts/me/")
+    return json.loads(response.content)
+
+
+def get_user_clubs(user):
+    """Returns list of clubs that user is a member of"""
+    response = authenticated_request(user, "GET", "https://pennclubs.com/api/memberships/")
+    res_json = json.loads(response.content)
+    return res_json
+
+
+def get_club_info(user, club_code):
+    """Returns club information based on club code"""
+    response = authenticated_request(user, "GET", f"https://pennclubs.com/api/clubs/{club_code}/")
+    res_json = json.loads(response.content)
+    return {"name": res_json["name"], "image": res_json["image_url"], "club_code": club_code}
+
+
 def get_user_populations(user):
     """Returns the target populations that the user belongs to"""
 
-    school = None
-    # school = get_affiliation(user.email)
-    # checks if school is in the target population
-    school_id = (
-        TargetPopulation.objects.get(population=school).id
-        if TargetPopulation.objects.filter(population=school).exists()
-        else -1
+    user_info = get_user_info(user)
+
+    content = []
+    content.extend(
+        [
+            TargetPopulation.objects.get(
+                kind=TargetPopulation.KIND_YEAR, population=user_info["student"]["graduation_year"]
+            )
+        ]
+    )
+    content.extend(
+        [
+            TargetPopulation.objects.get(kind=TargetPopulation.KIND_SCHOOL, population=x["name"])
+            for x in user_info["student"]["school"]
+        ]
+    )
+    content.extend(
+        [
+            TargetPopulation.objects.get(kind=TargetPopulation.KIND_MAJOR, population=x["name"])
+            for x in user_info["student"]["major"]
+        ]
     )
-    # year = user.profile.expected_graduation.year if user.profile.expected_graduation else None
-    year = None
-    # checks if year is in the target population
-    year_id = (
-        TargetPopulation.objects.get(population=year).id
-        if TargetPopulation.objects.filter(population=year).exists()
-        else -1
+    content.extend(
```

Comment:
Just curious: if the user satisfies more than one criterion (eg: targeted school & targeted major), will the same `TargetPopulation` be added twice to `content`?

Reply:
Oh nvm I think I got it: according to the current design, each TargetPopulation object can only specify one and only one kind. Could you confirm that my understanding is correct? @judtinzhang